### PR TITLE
Docs: First version of doc blocks migration guide

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: N/A

## What I did

Wrote migration guide for doc blocks, at least the ones merged so far. Added a couple deprecation warnings that were missing.

## How to test

Read the guide. Try doing deprecated stuff.

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [x] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -743,36 +747,102 @@ Instead, use `export const parameters = {};` and `export const decorators = [];`
...
+- Most other props controlling rendering of blocks now correspond precisely to the parameters for that block [defined for autodocs above](#autodocs-changes).
+
+##### Meta block
+
+The primary change of the `Meta` block is the ability to attach to CSF files with `<Meta of={}>` as described above.
+
+##### Description block, `parameters.notes` and `parameters.info`
+
+In 6.5 the Description doc block accepted a range of different props, `markdown`, `type` and `children` as a way to customize the content.
+The props have been simplified and the block now only accepts an `of` prop, which can be a reference to either a CSF file, a default export (meta) or a story export, depending on which description you want to be shown. See TDB DOCS LINK for a deeper explanation of the new prop.
```
Comment (shilman) [inline MIGRATION.md:841]: what does it mean to reference "a CSF file" in this context? `import * from Foo.stories.js`?

**Review by shilman — APPROVED** [review]
This is wonderful @tmeasday -- great job. Are we going to have a reference somewhere for all of the deprecated props/parameters and the rationale and what you should use instead? Or are we distributing that throughout the code in the form of deprecation warnings only?

**Comment by tmeasday** [comment]
I think I did mention all the deprecated props/parameters (excepting canvas) but I agree we could justify the components/subcomponents change more. I'm sort of hoping you'll maybe have some prose here?

File: MIGRATION.md [hunk 2]
```diff
@@ -743,36 +747,102 @@ Instead, use `export const parameters = {};` and `export const decorators = [];`
...
+
+#### Doc Blocks
+
+Additionally to changing the docs information architecture, we've updated the API of the doc blocks themselves to be more consistent and future proof.
+
+**General changes**
+
+- Each block now uses `of={}` as a primary API -- where the argument to the `of` prop is a CSF or story *export*.
+
+- When you've attached to a CSF file (with the `Meta` block, or in Autodocs), you can drop the `of` and the block will reference the first story or the CSF file as a whole.
```
Comment (JReinhold) [inline MIGRATION.md:830]: When reading this I would expect `Story` without an `of` to select the primary story in an attached file, but currently `Story` throws in that scenario. What's correct here?

File: MIGRATION.md (hunk 1 above)
Comment (JReinhold) [reply MIGRATION.md:841]: yes, or to be more precise `import * as FooStories from Foo.stories.js`

File: MIGRATION.md (hunk 2 above)
Comment (tmeasday) [reply MIGRATION.md:830]: I'm happy with either, simple enough to change. WDYT @shilman?

Comment (tmeasday) [reply MIGRATION.md:830]: Will do in a separate PR!

Comment (tmeasday) [reply MIGRATION.md:830]: https://github.com/storybookjs/storybook/pull/20765

**Pull request merged**